### PR TITLE
Chigusa Hydroelectric Dam - A damn big port.

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Ports the Trijent map from CM into TGMC with very minor map alterations. Provided this PR is merged, I'll use it as a base and slowly alter it to be more TGMC friendly.
I'm going to post the answers to a few questions here just to save time.

**Q:** Sounds good but how many CM assets does this use?
**A:** Some asphalt textures and one road decal. Everything else is stock TGMC or made by me.
**Q:** Can we do this? 
**A:** CM licenses their art under Creative Commons 3.0 BY-SA license, which allows for free use of art assets. If it somehow raises a fuss it's simple to redraw the small amount of CM textures this PR uses. In the event the map itself receives an official complaint, or a maintainer is uncomfortable I'll redo the entire thing to be more of a homage. (Which is probably what it will be post rework anyway.)
**Q:** Will I die to polluted water crossing a river?
**A:** Toxic water is not a feature that I ported, water filtration is nonfunctional and only cosmetic. 
**Q:** Will I see this map in rotation? Trijent _is_ pretty damn large...
**A.** Probably not frequently at first, the long term plan is to reduce it in size until it can comfortably fit 70-120 players. Even post-merge consider this map under active development.

## Why It's Good For The Game

More maps = more variety, provides a high-pop map for those times we go over 100 players. Gives us a kickass place to ride motorbikes and have HvH events. 

## Changelog
:cl:
add: Added Shigusa Hydroelectric Dam, a map for more than 100 players.
imageadd: Added a new couple icons for shallow water transitions, broken generators, and stairs.
code: Cleaned up some unused turfs.
/:cl:

### CODE REVIEW

**Comment by vilereaver41** [comment]
this is a disaster waiting to happen during the first few days

**Review by BraveMole — COMMENTED** [review]
So some remarks, i haven't reviewed anything yet, i'll probably do it tonight :

- Have you checked https://github.com/tgstation/TerraGov-Marine-Corps/pull/8104 ? Check that you are following these, we are just formalizing what was already enforced (more or less). If you have question about ai nodes, feel free to ping me on discord
- It's a pity that your map is only for over 90 players, because we don't have this pop. Consider shortening the map and making it medium-high pop (above 50-70), or it will never be played and it's a shame. Ultimatly, it's your map, so you do whatever you want
- Give credit to the author of the map, at least the orginal one (link the cm pr if you find, if you don't find the pr, just try to find the name of the author and ping him, it's always nice to get some credit)

Overall, very very nice to see a new map, good work

**Comment by RipGrayson** [comment]
> So some remarks, i haven't reviewed anything yet, i'll probably do it tonight :
> 
> * Have you checked [Map contrib guidelines added #8104](https://github.com/tgstation/TerraGov-Marine-Corps/pull/8104) ? Check that you are following these, we are just formalizing what was already enforced (more or less). If you have question about ai nodes, feel free to ping me on discord
I'll look into AI nodes after I fix the compile map errors.
> * It's a pity that your map is only for over 90 players, because we don't have this pop. Consider shortening the map and making it medium-high pop (above 50-70), or it will never be played and it's a shame. Ultimatly, it's your map, so you do whatever you want
This is something of a base, once it's in I'll work towards making the map more friendly towards our level of population.
> * Give credit to the author of the map, at least the orginal one (link the cm pr if you find, if you don't find the pr, just try to find the name of the author and ping him, it's always nice to get some credit)
Trijent predates TGMC by a bit (we actually have early versions of some of the prerelease assets still floating around) and was made by a team of CM mappers while CM was still closed source. I was unable to find any information on who worked on it, and there's no gitlab PR to credit the original submitter.

**Comment by BraveMole** [comment]
Undraft when ready for review

**Comment by RipGrayson** [comment]
This should be ready for review.

**Review by BraveMole — CHANGES_REQUESTED** [review]
Overall, good work, i'm really exciting about playing on this map!

image
 I'm afraid this crash site is out of the map? 
 
 
image

This is symptomatic of your nodes placement. It's not how you should do it, this will confuse ai the hell out. Please contact me on discord on how to place those.

image

What is this?

The LV needs to be marked (lv1, lv2) and also need to have shuttle computer. I suggest you use the landmarks one, so i won't have to change them if i make this map hvh compatible.

image

in all this area, xeno will be unable to build before shuttle. It's too big. Take into account the anti-wall spam protection.

image

Same remark, the areas need improvement here. I don't see any apc on the lzs? Fob should have their own area, there should be small areas around it, and then the rest is fine

image

Ugly as hell, use our catwalks

image

Weird patch of marking

image
 
this part feels really empty compared to the rest, can you fill it with stuff? Or just delete that part, the map is already huge.

image

This is too much of a dead end for me, add an exit on the south/north or just delete

image

Try to lower the amount of gens on the map to around 10-15. Also, it might be interesting to have two small engie part rather than one, it gives more objectives.

I see no intelligence computer, i know no one is make those missions but still, add a bunch.

**Comment by RipGrayson** [comment]
> Overall, good work, i'm really excited about playing on this map!
> 
> image
> I'm afraid this crash site is out of the map?
> 
Fixed
> image
> 
> This is symptomatic of your nodes placement. It's not how you should do it, this will confuse ai the hell out. Please contact me on discord on how to place those.
> 
Yeah I'll do that in a bit.
> image
> 
> What is this?
> 
I think CM has a mechanic where you can pilot escape pods off the planet. They stick the pods in there during transit for some reason, probably a lack of TG shuttle system. I removed the rooms, they have no use here.
> The LV needs to be marked (lv1, lv2) and also need to have shuttle computer. I suggest you use the landmarks one, so i won't have to change them if i make this map hvh compatible.
> 
> image
> 
In their infinite wisdom CM placed the shuttle computers north of LZ1 (the north one) in a small building, and just east of LZ2 in a small office directly adjacent. I didn't bother changing the positions but I did make sure both are accessible and protected by shutters. I can easily move both of them, I was just pointing out that both LZs technically have computers.
> in all this area, xeno will be unable to build before shuttle. It's too big. Take into account the anti-wall spam protection.
> 
Reduced the size by about half.
> image
> 
> Same remark, the areas need improvement here. I don't see any apc on the lzs? Fob should have their own area, there should be small areas around it, and then the rest is fine
> 
Come to think of it, apparently CM has no APCs on their LZs. Weird design decision. I added one to both and expanded the external FOB areas.
> image
> 
> Ugly as hell, use our catwalks
> 
Done.
> image
> 
> Weird patch of marking
> 
Fixed.
> image
> 
> this part feels really empty compared to the rest, can you fill it with stuff? Or just delete that part, the map is already huge.
> 
This area used to be filled with filtration equipment, since my philosophy was to avoid using as much CM art as possible I just blanked the areas where spritework was. I'll either replace the area, or delete it entirely soon.
> image
> 
> This is too much of a dead end for me, add an exit on the south/north or just delete
> 
Done.
> image
> 
> Try to lower the amount of gens on the map to around 10-15. Also, it might be interesting to have two small engie part rather than one, it gives more objectives.
> 
Believe it or not during testing it takes that many gens to fully power Trijent. I'll reduce it down to 12 and see how it does.
> I see no intelligence computer, i know no one is make those missions but still, add a bunch.
>
Added 25 intel computers in various areas of the map.

**Comment by AndroidMK2** [comment]
Seems very epic would definetly want to play on it 😳

**Comment by bowlofcereal** [comment]
This is actually fucking cool dude.

File: code/__DEFINES/minimap.dm
Comment (BraveMole) [inline code/__DEFINES/minimap.dm:41]: you are not using this. This is out of scope for the PR anyway, please remove

**Review by BraveMole — CHANGES_REQUESTED** [review]
Solid work, appart from these two remarks and the nodes it's fine for me.

image

You need shutters all around the lz, because marines are dumb and will PC the rock to get out

image

Weird empty space

**Comment by AnCopper** [comment]
Trijent without the poop water is like bread without butter, smh..

**Comment by RipGrayson** [comment]
> Trijent without the poop water is like bread without butter, smh..

If people want the toxic water mechanics back I'll look into restoring it post merge. I'm not currently sensing a great deal of demand for it though.

**Comment by mrhugo13** [comment]
Need to add miners, without it things economy side get a bit ehhhhhhhhhh.

File: _maps/map_files/desertdam/desertdam.dmm
Comment (BraveMole) [inline _maps/map_files/desertdam/desertdam.dmm]: It's not allowed to make vv on map objects

**Review by BraveMole — CHANGES_REQUESTED** [review]
So i see you added miners, good.

Unfortunatly, the map is way too big for our map, so i propose two options: 

- We freeze this map until we get vehicles, which could take from 1 to 3 months (apc, things like that). And then we give a ton of free vehicles, and a lot of already spawned tunnels for xeno
- You cut 30-50% of the map

**Comment by RipGrayson** [comment]
> So i see you added miners, good.
> 
> Unfortunatly, the map is way too big for our map, so i propose two options:
> 
> * We freeze this map until we get vehicles, which could take from 1 to 3 months (apc, things like that). And then we give a ton of free vehicles, and a lot of already spawned tunnels for xeno
> * You cut 30-50% of the map

I'm no illusion that this map is perfect, it's a map made for a different codebase with minimal changes.
Unfortunately I cannot cut 50% of the map quickly, it would take me at least a month to do it. This map is a base, I was planning on incrementally implementing size and balance changes after merge, collecting feedback on each change.

I guess if you want we could just add the map and let it be inaccessible to all but admins while I work on improving it.

**Comment by BraveMole** [comment]
> I'm no illusion that this map is perfect, it's a map made for a different codebase with minimal changes.
> Unfortunately I cannot cut 50% of the map quickly, it would take me at least a month to do it. This map is a base, I was planning on incrementally implementing size and balance changes after merge, collecting feedback on each change.
> 
> I guess if you want we could just add the map and let it be inaccessible to all but admins while I work on improving it.

No rush, since it's not really votable due to the pop config, i can merge without issue. Honestly i think the vehicle idea might make this map really unique, and tivi is on that

**Review by BraveMole — CHANGES_REQUESTED** [review]
I'll merge when the plateform problem is fixed.

Code wise, it shouldn't be that hard, you could do this in another pr. Otherwise, you can remove all plateform but that would be a shame;

Another note, this miner is too close to the lz
image
Remove it please

**Comment by RipGrayson** [comment]
Well platforms have been fixed, I don't think there's anything else in the way of preventing a merge.

**Review by BraveMole — APPROVED** [review]

**Pull request merged**
